exporter_test.go

```diff
@@ -1,55 +1,129 @@
+		{"positive expect regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", "[oO][kK]"), "probe_success 1", http.StatusOK},
+		{"negative expect regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", "fail"), "probe_success 0", http.StatusOK},
+		{"empty expect regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", ""), "probe_success 1", http.StatusOK},
+		{"invalid expect regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", "*"), "probe_success 0", http.StatusOK},
+		{"positive fail on regexp test", ts["simple"].URL, "", urlValue("http_fail_on_regexp", "[oO][kK]"), "probe_success 0", http.StatusOK},
+		{"negative fail on regexp test", ts["simple"].URL, "", urlValue("http_fail_on_regexp", "fail"), "probe_success 1", http.StatusOK},
+		{"empty fail on regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", ""), "probe_success 1", http.StatusOK},
+		{"confusing regexp test", ts["simple"].URL, "", urlValue("http_expect_regexp", "ok", "http_fail_on_regexp", "ok"), "probe_success 0", http.StatusOK},
+		{"positive basic auth test", ts["basicAuth"].URL, "", urlValue("http_basic_auth_username", testUsername, "http_basic_auth_password", testPassword), "probe_success 1", http.StatusOK},
+		{"negative basic auth test", ts["basicAuth"].URL, "", urlValue("http_basic_auth_username", "myUsername", "http_basic_auth_password", "myPassword"), "probe_success 0", http.StatusOK},
+		{"only username basic auth test", ts["basicAuth"].URL, "", urlValue("http_basic_auth_username", testUsername), "probe_success 0", http.StatusOK},
+		{"validate that login is not possible for user", ts["basicAuth"].URL, "", urlValue("http_basic_auth_username", "myUser", "http_basic_auth_password", "myPassword", "http_valid_status_codes", "401"), "probe_success 1", http.StatusOK},
 	}
 	for _, tt := range handlerTests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			req := httptest.NewRequest("GET", "http://exporter/probe?target="+tt.target+"&module="+tt.module, nil)
+			testURL := "http://exporter/probe?target=" + tt.target + "&module=" + tt.module
+			if len(tt.overwrites) > 0 {
+				testURL = testURL + "&" + tt.overwrites.Encode()
+			}
+			fmt.Printf("Using URL: %s\n", testURL)
+			req := httptest.NewRequest("GET", testURL, nil)
 			w := httptest.NewRecorder()
 			Handler(w, req)
 			resp := w.Result()
 			body, err := ioutil.ReadAll(resp.Body)
 			assert.NoError(t, err)
 
-			assert.Equal(t, tt.expectedStatus, resp.StatusCode)
+			assert.Equal(t, tt.expectedStatus, resp.StatusCode, fmt.Sprintf("Expected status code %v, but got %v", tt.expectedStatus, resp.StatusCode))
 			assert.Contains(t, string(body), tt.bodyContains)
 		})
 	}
 }
+
+func urlValue(keyValue ...string) url.Values {
+	if len(keyValue)%2 != 0 {
+		panic("Expected key-value pairs as arguments, but got uneven amount")
```

Comment:
How does a panic behave in a test? If just the affected subtest fails, I'm fine with it but it should not affect the other tests.

Reply:
Changed it, so that a wrong usage will fail the relevant subtest